### PR TITLE
Allow for custom install prefix

### PR DESCRIPTION
**Pull request by 06kellyjac**
Allow for custom install prefix
Also create the dir if it doesn't exist and ensure 755 perms

### CODE REVIEW

**Comment by 06kellyjac** [comment]
@stanley-cheung @sampajano Please can I get a review on this

This is required for us to easily package this for nixpkgs https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/tools/protoc-gen-grpc-web/default.nix#L16-L25

File: javascript/net/grpc/web/Makefile [hunk 1]
```diff
@@ -16,14 +16,15 @@ CXX ?= g++
...
+PREFIX ?= /usr/local
 
 all: protoc-gen-grpc-web
 
 protoc-gen-grpc-web: grpc_generator.o
 	$(CXX) $^ $(LDFLAGS) -o $@
 
 install: protoc-gen-grpc-web
-	install protoc-gen-grpc-web /usr/local/bin/protoc-gen-grpc-web
+	install -Dm755 protoc-gen-grpc-web $(PREFIX)/bin/protoc-gen-grpc-web
```
Comment (sampajano) [inline javascript/net/grpc/web/Makefile:27]: I don't think `-Dm755` work on macOS (BSD) version of `install`..

Is there a way you can make this work for both? :)

Comment (stanley-cheung) [reply javascript/net/grpc/web/Makefile:27]: Can we leave this `-Dm755` out and just use the system default? the `PREFIX` change looks good.

Comment (06kellyjac) [reply javascript/net/grpc/web/Makefile:27]: I can either drop it or add a `mkdir -p $(PREFIX)` above `install`

Comment (sampajano) [reply javascript/net/grpc/web/Makefile]: Yeah this is cool with me.. But i'll defer to @stanley-cheung's for approval :)

Thanks for the change :)

**Review by stanley-cheung — APPROVED** [review]

**Pull request merged**